lib/client/mfa/cli.go

```diff
@@ -185,17 +185,12 @@ func (c *CLIPrompt) Run(ctx context.Context, chal *proto.MFAAuthenticateChalleng
 		fmt.Fprintf(c.writer(), msg, strings.Join(availableMethods, ", "), strings.Join(chosenMethods, " and "))
 	}
 
-	// DELETE IN v18.0 after TOTP session MFA support is removed (codingllama)
-	// Technically we could remove api/mfa.WithPromptChallengeExtensions along
-	// with this, as it's likely its only use, although arguably keeping it could
-	// prove useful.
-	usageSessionMFA := c.cfg.Extensions.GetScope() == mfav1.ChallengeScope_CHALLENGE_SCOPE_USER_SESSION
-	if promptOTP && usageSessionMFA {
-		fmt.Fprint(c.writer(), "\nWARNING: Starting with Teleport 18, OTP will not be accepted for per-session MFA.\n\n")
-	}
+	isPerSessionMFA := c.cfg.Extensions.GetScope() == mfav1.ChallengeScope_CHALLENGE_SCOPE_USER_SESSION
 
+	// We should never prompt for OTP when per-session MFA is enabled, but MFA prompts are
+	// unrestricted otherwise
 	switch {
-	case promptOTP && promptWebauthn:
+	case !isPerSessionMFA && promptOTP && promptWebauthn:
```

Comment:
Instead of including `isPerSessionMFA` within `switch` branches, I wonder if it wouldn't be clearer to add an extra check for `isPerSessionMFA` within each branch that uses `promptOTP`.

It introduces a tiny bit of duplication, but I think the code is easier to read this way – `switch` would then be reserved only for different prompts and the person reading the code wouldn't need to consider the value of `isPerSessionMFA` to understand which branch of the `switch` the execution will go to.

Reply:
The thing is though, with how the code is right now, if `promptOTP` and `promptWebauthn` are true but the prompt is for per-session MFA, the execution here will go to the `case promptWebauthn` branch. I suppose this behavior could still be kept even if `isPerSessionMFA` checks were inside `case prompOTP && promptWebauthn` instead of being within the case itself.

Reply:
Yeah I went back and forth on where best to inject the `isPerSessionMFA` check. I just pushed a change that disables `promptOTP` if there's another MFA method available during per-session MFA, which I think does a better job of conveying what's happening :crossed_fingers: